firedrake/slate/slac/utils.py

```diff
@@ -136,6 +152,224 @@ def visit_Symbol(self, o, *args, **kwargs):
         return SymbolWithFuncallIndexing(o.symbol, o.rank, o.offset)
 
 
+# singledispatch for class functions
+def classsingledispatch(func):
+    dispatcher = singledispatch(func)
+
+    def wrapper(*args, **kw):
+        return dispatcher.dispatch(args[1].__class__)(*args, **kw)
+
+    wrapper.register = dispatcher.register
+    update_wrapper(wrapper, func)
+    return wrapper
+
+
+class SlateTranslator():
+    """Multifunction for translating UFL -> GEM.  """
```

Comment:
Obviously not!

Reply:
Adressed in 418eaea8e307de39983cf9eb60caa8e01d266212